pkg/httphelper/httphelper.go

```diff
@@ -0,0 +1,108 @@
+package httphelper
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// TraceResponseWriter provides a wrapper to write data and header
+type TraceResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+// WriteHeader writes the header to the response
+func (w *TraceResponseWriter) WriteHeader(code int) {
+	w.statusCode = code
+	w.ResponseWriter.WriteHeader(code)
+}
+
+// Write writes the body of the response
+func (w *TraceResponseWriter) Write(data []byte) (int, error) {
+	size, err := w.ResponseWriter.Write(data)
+	w.size += size
+	return size, err
+}
+
+// Metrics is responsible for holding the metrics for Prometheus
+type Metrics struct {
+	HTTPRequestDuration *prometheus.HistogramVec
+	HTTPResponseSize    *prometheus.HistogramVec
+	ErrorRate           *prometheus.CounterVec
+}
+
+// NewMetrics is a constructor for Metrics
+func NewMetrics(namespace string) *Metrics {
+	m := &Metrics{
+
+		HTTPRequestDuration: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: namespace,
+				Name:      "http_request_duration_seconds",
```

Comment:
We already have https://github.com/kubernetes/test-infra/blob/master/prow/metrics/http.go, let's use it.

Reply:
do you want me to move the tests in http_helper to prow or are they not necessary?

Reply:
Also the buckets for httpDuration will probably be different from what they were earlier if we use the prow/metric, since it allows only for powers of 2 but right now it is - {0.0005, 0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2}.
Will that be ok?

Reply:
Yes, those are fine.